lib/Connector/XiboAudienceReportingConnector.php

```diff
@@ -216,19 +216,27 @@ public function onRegularMaintenance(MaintenanceRegularEvent $event)
                     $entry['campaignStart'] = $adCampaignCache[$parentCampaignId]['start'];
                     $entry['campaignEnd'] = $adCampaignCache[$parentCampaignId]['end'];
                 } else {
-                    $parentCampaign = $this->campaignFactory->getById($parentCampaignId);
-                    if ($parentCampaign->type == 'ad') {
-                        $adCampaignCache[$parentCampaignId]['type'] = $parentCampaign->type;
-                    } else {
-                        $listCampaignCache[$parentCampaignId] = $parentCampaignId;
-                        continue;
-                    }
+                    // Get Campaign
+                    try {
+                        $parentCampaign = $this->campaignFactory->getById($parentCampaignId);
 
-                    if (!empty($parentCampaign->getStartDt()) && !empty($parentCampaign->getEndDt())) {
-                        $adCampaignCache[$parentCampaignId]['start'] =  $parentCampaign->getStartDt()->format(DateFormatHelper::getSystemFormat());
-                        $adCampaignCache[$parentCampaignId]['end'] = $parentCampaign->getEndDt()->format(DateFormatHelper::getSystemFormat());
-                        $entry['campaignStart'] = $adCampaignCache[$parentCampaignId]['start'];
-                        $entry['campaignEnd'] = $adCampaignCache[$parentCampaignId]['end'];
+                        if ($parentCampaign->type == 'ad') {
+                            $adCampaignCache[$parentCampaignId]['type'] = $parentCampaign->type;
+                        } else {
+                            $listCampaignCache[$parentCampaignId] = $parentCampaignId;
+                            continue;
+                        }
+
+                        if (!empty($parentCampaign->getStartDt()) && !empty($parentCampaign->getEndDt())) {
+                            $adCampaignCache[$parentCampaignId]['start'] =  $parentCampaign->getStartDt()->format(DateFormatHelper::getSystemFormat());
+                            $adCampaignCache[$parentCampaignId]['end'] = $parentCampaign->getEndDt()->format(DateFormatHelper::getSystemFormat());
+                            $entry['campaignStart'] = $adCampaignCache[$parentCampaignId]['start'];
+                            $entry['campaignEnd'] = $adCampaignCache[$parentCampaignId]['end'];
+                        }
+                    } catch (\Exception $exception) {
+                        $event->addMessage(__('Error caching ad/list campaign:'. $exception->getMessage()));
+                        $this->getLogger()->error('onRegularMaintenance: caching ad/list campaign failed' . $exception->getMessage());
```

Comment:
If there are 5000 records all for this campaign, we're going to log this 5000 times and have a really long event message as well.